### DISCUSSION TITLE
Looking help/answers on how to correctly deploy to aws

### DISCUSSION THREAD

**Discussion by dougieled** [hagopj13/node-express-boilerplate#236] [Q&A] [answered]

**Comment by anindosarker** [comment]
Following here. I'm also struggling on how to deploy this project on AWS elastic beanstalk with the ci/cd pipelines

**Reply by amadebusuyi** [reply]
@anindosarker What part of the process have you completed?

**Reply by anindosarker** [reply]
Thanks for the reply. However I was able to deploy it on my own using GitHub actions runner as CI/CD and pm2 to run the project. The `yarn start` command doesn't start the pm2 process as intended. So, I installed pm2 on my ec2 instance and manually ran the pm2 start command instead of using yarn.

This video was really helpful throughout the process https://youtu.be/cgWXQqL-ZU8?si=5mevZFUQ_VIUr-6X

**Reply by amadebusuyi** [reply]
Great.

**Comment by amadebusuyi** [comment]
I struggled with this for a while as well.

**How I resolved the issue:** I changed the package manager in the Dockerfile to NPM.

**Comment by anindosarker** [comment, marked as answer]
This instructions are for EC2. However you'll understand the basics if you follow these 2 videos.

Without docker deployment:
https://youtu.be/cgWXQqL-ZU8?si=5mevZFUQ_VIUr-6X

With docker :
https://youtu.be/rRes9LM-Jh8?si=cVPFabEWsjWVguaa


After this, you should

- Understand the basics
- Modify the GitHub actions yml file according to this template, set up the branches for deployment etc
- Add running commands

Test it out
